### PR TITLE
Removal of deprecated classes, cleanup of address view and implementation of spliterator

### PR DESCRIPTION
**Pull request by no2chem**
This pull request removes support for Replex, which will not be available in the 1.0 release.
It also removes support for runtime instrumentation (after this PR, the depedency on
ByteBuddy can be removed).

The annotationProcessor has been updated to support inner static classes (anonymous classes
were never supported under ByteBuddy). 

Deprecated APIs in AddressSpaceView have been removed (such as holeFill), and deprecated
methods in ObjectView and LogUnitClient have also been removed.

A new spliterator implementation reduces memory pressure during traversal of a large
stream for the first time.

### CODE REVIEW

**Comment by no2chem** [comment]
Partially resolves #566

**Comment by Maithem** [comment]
@no2chem do you need the split iterator implementation to remove the deprecated classes and replex?
would it be hard to separate the two?

File: runtime/src/main/java/org/corfudb/runtime/view/AbstractReplicationView.java [hunk 1]
```diff
@@ -1,129 +0,0 @@
-package org.corfudb.runtime.view;
```
Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/view/AbstractReplicationView.java]: I understand we are focusing this branch on quorum replication only. But why not retain the abstractions and interfaces that allow flexible replication schemes in the future?

File: runtime/src/main/java/org/corfudb/runtime/view/ObjectBuilder.java [hunk 2]
```diff
@@ -1,14 +1,14 @@
 package org.corfudb.runtime.view;
 
 import com.google.common.reflect.TypeToken;
+import com.sun.xml.internal.bind.v2.model.runtime.RuntimeBuiltinLeafInfo;
```
Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/view/ObjectBuilder.java]: mistakenly inserted?

File: runtime/src/main/java/org/corfudb/runtime/view/AbstractReplicationView.java (hunk 1 above)
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/AbstractReplicationView.java]: But we do, this code was marked deprecated, and the flexibility is retained in protocols

File: runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java [hunk 3]
```diff
@@ -37,7 +37,7 @@ public Object getPayload(CorfuRuntime runtime) {
                     if (data == null) {
                         this.payload.set(null);
                     } else {
-                        ByteBuf copyBuf = Unpooled.copiedBuffer(data);
+                        ByteBuf copyBuf = Unpooled.wrappedBuffer(data);
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java]: I don't think so. The memory leaks were coming from pooled memory.

Comment (no2chem) [inline runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java]: not a memory leak, but an unnecessary copy, I believe

**Comment by no2chem** [comment]
@Maithem will address protobufs in a separate PR

**Comment by Maithem** [comment]
@no2chem dont worry, I can do it. Im working on another cleanup patch.

**Pull request merged**